### ISSUE TITLE
`docker-machine` is deprecated, we need another tool for autoscaling

### ISSUE THREAD

**Issue by gromain** [OpenDroneMap/ClusterODM#88]
`docker-machine` is being deprecated and starts to be unavailable from repositories. It's also not compatible with the latest docker version. See https://github.com/docker/machine/issues/4537 and https://github.com/docker/roadmap/issues/245 .

One alternative would be to move towards [Terraform](https://learn.hashicorp.com/collections/terraform/cli). There is a [provider for Scaleway](https://github.com/scaleway/terraform-provider-scaleway), so we wouldn't lose functionality. Is there any interest in moving towards this solution?

### What is the problem?
`docker-machine` being deprecated.

### What should be the expected behavior?
Not losing the autoscale!

### How can we reproduce this? (What steps did you do to trigger the problem? If applicable, please include multiple screenshots of the problem! Be detailed)
Try installing docker-machine on a recent OS.

**Comment by gromain** [comment]
A python alternative could be libcloud (I know this tool is written in JS, just mentioning this for posterity): https://libcloud.apache.org/

**Comment by gromain** [comment]
Ah yes, this is good then!
Maybe we can just update the Readme to point to this new repo, instead of towards https://docs.docker.com/machine/install-machine/ . It was really confusing for me!

**Comment by gromain** [comment]
I've fixed the documentation, Dockerfile and docker-compose files.
This will do for now, but I suggest we keep this open to make sure the tool is not completely abandoned by Gitlab and we keep looking for another solution.

**Comment by pierotofy** [comment]
Thanks. I've reverted the docker-compose file changes since not everyone uses the autoscaling functionality (perhaps those changes should be part of a separate docker-compose file).

**Comment by gromain** [comment]
Oh I see, I thought the docker-composer file here was only for the autoscaler (since this was in the autoscaler section of the readme too and there is specific instructions about starting it up with one child nodeodm).
I'll create a separate one and update the readme accordingly.

**Comment by theoway** [comment]
> A python alternative could be libcloud (I know this tool is written in JS, just mentioning this for posterity): https://libcloud.apache.org/

Can we not migrate to Terraform? It is very widely supported.

**Comment by ArthurD3nt** [comment]
Hello everyone,
I haven’t looked at the details of the migration to these platforms yet but I made a list of plausible alternatives. 

## The literature

I did some reading of the [issue](https://github.com/docker-archive-public/docker.machine/issues/4537) that announces the start of “”maintenance mode”” for docker-machine. 
First of all they state that maintenance mode means that they will still do bug fixes but not implement new features. This is clearly false and misleading as it’s clear from the amount and the nature of the issues open that it is clearly abandoned in favor of their closed source enterprise products.

One of the maintainers [states](https://github.com/docker-archive-public/docker.machine/issues/4537#issue-341148889) that the alternatives should be:

- Windows: Docker desktop
- MacOS: Docker desktop
- Linux: [InfraKit](https://github.com/docker/infrakit)

InfraKit, as stated by this gentleman [here](https://github.com/docker-archive-public/docker.machine/issues/4537#issuecomment-587271780), is a tool for system builders. It is very complex and way less usable than docker-machine. Also, as it’s in their “”maintenance mode”” as well, I don’t really see the point in switching from one deprecated library to another.

TLDR:Docker as a company wanted to sell more enterprise solutions and these open source ones were menacing their business model so they pulled the plug.

Gitlab out of necessity continued to update his fork of docker-machine used internally for their CI/CD pipeline but they too are migrating away from it.

## The alternatives

I guess we would want to keep support for these could providers:

- DigitalOcean
- Hetzner
- Scaleway
- Amazon Web Services

| **Provisioning tool** | **DigitalOcean** | **Hetzner** | **Scaleway** | **AWS** |
| --- | --- | --- | --- | --- |
| [Terraform](https://developer.hashicorp.com/terraform) | ✅ | ✅ | ✅ | ✅ |
| [Libcloud](https://libcloud.apache.org/) | ✅ | ❌ | ✅ | ✅ |
| [OpenTofu](https://opentofu.org/) | ✅ | ✅ | ✅ | ✅ |

Terraform has the benefit of being the most widely used and basically a industry standard.
